src/bootstrap.js

```diff
@@ -216,7 +198,16 @@ export function bootstrap(initialConfig) {
 			version: version,
 			autoConfirm: false,
 			analytics: true,
-			bufferEvents: config.requireConsent
+			bufferEvents: config.requireConsent,
+			hooks: {
+				beforeOptions: function(opts) {
+					return (previewCid)
+						? objectAssign(opts, {
+							url: opts.url + '?previewcid=' + previewCid
+						})
+						: opts;
+				}
+			}
 		};
```

Comment:
The only option I see -- that's to add `previewCid` field or `queryParams` object directly to EvolvClient options. But the hooks approach looks like a more universal one.

Also that could be possible to add `previewCid` to `client.context` via `client.initialize()` or `client.context.set()` (i.e. similar to `uid` and `sid`), but I'm not sure if modifying context is a right way as it emits `context.updated` events on every change.

Reply:
Is the question that you're answering my question from [here](https://github.com/evolv-ai/javascript-sdk/pull/104#issue-1086340452) about whether the hooks approach is ideal?

Reply:
yep, the hooks approach looks fine to me.